### PR TITLE
[#8546] Style statistics

### PR DESCRIPTION
**Pull request by sevfurneaux**
**Describe your changes**
This PR styles the statistics box

Screenshot 2024-11-18 at 15 39 38

**Tasks**
- [x] PR name contains story or task reference
- [ ] Steps to recreate and test the changes
- [ ] Documentation (docs and inline)
- [ ] Tests (including n+1 and django_assert_num_queries where applicable)
- [x] Changelog

### CODE REVIEW

File: meinberlin/assets/scss/components_user_facing/_live_questions.scss [hunk 1]
```diff
@@ -7,3 +7,29 @@
...
+
+.live_question__progress-text {
+    padding-left: 8px;
+    padding-right: 8px; 
+    position: relative;
+    z-index: 1;
+}
+
+.live_question__progress-percentage {
+    font-weight: bold;
```
Comment (vellip) [inline meinberlin/assets/scss/components_user_facing/_live_questions.scss:34]: Can you use `live-question` instead of `live_question`? Is more BEM-like and already defined like that in the rest of the file.

File: meinberlin/assets/scss/components_user_facing/_live_questions.scss [hunk 2]
```diff
@@ -7,3 +7,29 @@
...
+    background-color: $gray-lighter;
+    border-radius: 10px;
+    height: 100%;
+    position: absolute;
+    z-index: 0;
+}
+
+.live_question__progress-text {
+    padding-left: 8px;
+    padding-right: 8px;
```
Comment (vellip) [inline meinberlin/assets/scss/components_user_facing/_live_questions.scss:28]: can you use `em` or `rem` more, where it makes sense please? I think we prefer that over px, unless its some specific number or similar.

File: meinberlin/react/livequestions/StatisticsBox.jsx
Comment (vellip) [inline meinberlin/react/livequestions/StatisticsBox.jsx:1]: Could you convert this to a functional component as well? Or did you plan to do that in another PR?

File: meinberlin/react/livequestions/StatisticsBox.jsx [hunk 3]
```diff
@@ -42,83 +42,79 @@ export default class StatisticsBox extends React.Component {
 
   render () {
     const questionAnsweredTag = django.gettext('Questions Answered')
+    const categoriesTag = django.gettext('Categories')
     const categoriesAnsweredTag = django.gettext('Affiliation Of Answered Questions')
```
Comment (vellip) [inline meinberlin/react/livequestions/StatisticsBox.jsx:46]: Could you move these outside the component, if possible? Thanks!

File: meinberlin/react/livequestions/StatisticsBox.jsx [hunk 4]
```diff
@@ -42,83 +42,79 @@ export default class StatisticsBox extends React.Component {
...
+        {this.props.categories.length > 0 && (
+          <>
+            <h2>{categoriesAnsweredTag}</h2>
+            <div className="modul-card card">
+              <h3>{categoriesTag}</h3>
+              {this.props.categories.map((category, index) => {
+                const countPerCategory = this.countCategory(category)
+                const style = { width: countPerCategory + '%' }
+                return (
+                  <div className="live_question__progress" key={index}>
```
Comment (vellip) [inline meinberlin/react/livequestions/StatisticsBox.jsx:59]: I know it's been like that in the code before, but could you change it to use something other than `index`?

**Review by vellip — COMMENTED** [review]
Nice job, I left just a few minor notes for you.

File: meinberlin/react/livequestions/react_questions_statistics.jsx [hunk 5]
```diff
@@ -1,13 +0,0 @@
-import StatisticsBox from './StatisticsBox'
```
Comment (sevfurneaux) [inline meinberlin/react/livequestions/react_questions_statistics.jsx:1]: @vellip I noticed that `<StatisticsBox />` was also rendered in react_questions_statistics.jsx, but react_questions_statistics.jsx isn't used anywhere or referenced in webpack.common.js, so have removed it.

File: meinberlin/react/livequestions/StatisticsBox.jsx (hunk 4 above)
Comment (sevfurneaux) [reply meinberlin/react/livequestions/StatisticsBox.jsx]: Thanks, I've now added `category_dict` here, as the keys are the `category` `id`s: https://github.com/liqd/a4-meinberlin/pull/5750/files#diff-2d2b8a5601ea372ea7c39a1e981837692b43614c93d46ab5c0a13f39c8196dd3R54

File: meinberlin/react/livequestions/StatisticsBox.jsx
Comment (sevfurneaux) [reply meinberlin/react/livequestions/StatisticsBox.jsx:1]: This is now converted.

**Comment by sevfurneaux** [comment]
> Nice job, I left just a few minor notes for you.

Thanks for the review, I've now made the changes 👍

**Review by vellip — APPROVED** [review]
